Base/src/test/java/io/deephaven/base/FileUtilsTest.java

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.base;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+
+import junit.framework.TestCase;
+import org.junit.Assert;
+import org.junit.rules.TemporaryFolder;
+
+public class FileUtilsTest extends TestCase {
+
+    public void testConvertToFileURI() throws IOException {
+        final TemporaryFolder folder = new TemporaryFolder();
+        folder.create();
+
+        final File someDir = folder.newFolder();
+        fileUriTestHelper(someDir.getPath(), true, someDir.toURI().toString());
+
+        final File someFile = folder.newFile();
+        fileUriTestHelper(someFile.getPath(), false, someFile.toURI().toString());
+
+        // Check if trailing slash gets added for a directory
+        final String expectedURI = "file:" + someDir.getPath() + "/path/to/directory/";
+        fileUriTestHelper(someDir.getPath() + "/path/to/directory", true, expectedURI);
+
+        // Check if multiple slashes get normalized
+        fileUriTestHelper(someDir.getPath() + "////path///to////directory", true, expectedURI);
+
+        // Check if multiple slashes in the beginning get normalized
+        fileUriTestHelper("////" + someDir.getPath() + "/path/to/directory", true, expectedURI);
+    }
+
+    private static void fileUriTestHelper(final String filePath, final boolean isDirectory,
+            final String expctedURIString) {
+        Assert.assertEquals(expctedURIString, FileUtils.convertToURI(filePath, isDirectory).toString());
+        Assert.assertEquals(expctedURIString, FileUtils.convertToURI(new File(filePath), isDirectory).toString());
+        Assert.assertEquals(expctedURIString, FileUtils.convertToURI(Path.of(filePath), isDirectory).toString());
+    }
+
+    public void testConvertToS3URI() throws URISyntaxException {
+        assertEquals("s3:/bucket/key", FileUtils.convertToURI("s3:/bucket/key", false).toString());
```

Comment:
Hmmm; aren't s3 URIs supposed to look like `s3://bucket/some/key/path`? How do we currently handle `s3:/bucket/key`?

Reply:
You are right, I did have a bad input and have fixed the test for now. 
Regarding `s3:/bucket/key`, will check and get back to you.

Reply:
The convertToURI code would return the same output for this input `"s3:/bucket/key/"`.

I tried fetching such a URI: `"s3:/dh-s3-parquet-test1/singleColFile.parquet"` and got the following error:
```
RuntimeError: java.lang.IllegalArgumentException: Invalid S3 URI: bucket not included: s3:/dh-s3-parquet-test1/singleColFile.parquet
	at software.amazon.awssdk.services.s3.S3Utilities.parseAwsCliStyleUri(S3Utilities.java:387)
	at software.amazon.awssdk.services.s3.S3Utilities.parseUri(S3Utilities.java:296)
	at io.deephaven.extensions.s3.S3SeekableChannelProvider.getReadChannel(S3SeekableChannelProvider.java:82)
	at io.deephaven.util.channel.CachedChannelProvider.getReadChannel(CachedChannelProvider.java:98)
	at io.deephaven.parquet.base.ParquetFileReader.<init>(ParquetFileReader.java:98)
	at io.deephaven.parquet.base.ParquetFileReader.create(ParquetFileReader.java:73)
```
So looks like it gets handled at the lower layer.